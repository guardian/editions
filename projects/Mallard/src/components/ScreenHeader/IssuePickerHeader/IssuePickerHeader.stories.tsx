import { color, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import React from 'react';
import { IssuePickerHeader } from './IssuePickerHeader';

storiesOf('IssuePickerHeader', module)
	.addDecorator(withKnobs)
	.add('Default', () => (
		<IssuePickerHeader title="Recent" subTitle="Editions" />
	))
	.add('With Different Title and SubTitle', () => (
		<IssuePickerHeader title="Australia" subTitle="Weekend" />
	))
	.add('With Header Styles', () => (
		<IssuePickerHeader
			title="Food"
			subTitle="Monthly"
			headerStyles={{
				backgroundColor: color('Background Colour', '#7D0068'),
				textColorPrimary: color('Text Colour Primary', '#007ABC'),
				textColorSecondary: color('Text Colour Secondary', '#F3C100'),
			}}
		/>
	));
