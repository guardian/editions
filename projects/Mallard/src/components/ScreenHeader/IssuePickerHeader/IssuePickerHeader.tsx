import React from 'react'
import { NavigationInjectedProps, withNavigation } from 'react-navigation'
import { SpecialEditionHeaderStyles } from 'src/common'
import { navigateToSettings } from 'src/navigation/helpers/base'
import { ScreenHeader } from '../ScreenHeader'
import { CloseButton } from 'src/components/Button/CloseButton'
import { SettingsButton } from 'src/components/Button/SettingsButton'
import { Header } from 'src/components/layout/header/header'
import { IssueTitle } from 'src/components/issue/issue-title'
import { styles } from 'src/components/styled-text'

const IssuePickerHeader = withNavigation(
    ({
        headerStyles,
        navigation,
        subTitle,
        title,
    }: {
        headerStyles?: SpecialEditionHeaderStyles
        subTitle?: string
        title: string
    } & NavigationInjectedProps) => (
        <Header
            alignment={'drawer'}
            onPress={() => navigation.goBack()}
            action={
                <CloseButton
                    accessibilityLabel="Close button"
                    accessibilityHint="Returns to the edition"
                    onPress={() => navigation.goBack()}
                />
            }
            leftAction={
                <SettingsButton
                    onPress={() => {
                        navigateToSettings(navigation)
                    }}
                />
            }
            headerStyles={headerStyles}
        >
            {title ? (
                <IssueTitle
                    title={title}
                    subtitle={subTitle}
                    titleStyle={styles.issueLightText}
                    overwriteStyles={headerStyles}
                />
            ) : null}
        </Header>
    ),
)

export { IssuePickerHeader }
