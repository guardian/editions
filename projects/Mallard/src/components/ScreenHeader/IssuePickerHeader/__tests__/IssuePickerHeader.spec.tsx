import React from 'react'
import TestRenderer, { ReactTestRendererJSON } from 'react-test-renderer'
import { IssuePickerHeader } from '../IssuePickerHeader'

jest.mock('react-navigation', () => ({
    withNavigation: (child: any) => child,
}))

describe('IssuePickerHeader', () => {
    it('should match the default style', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <IssuePickerHeader title="UK" subTitle="Daily" />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
    it('should match the default style with a subTitle', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <IssuePickerHeader title="Recent" subTitle="Editions" />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
    it('should match the altered style by the prop headerStyles', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <IssuePickerHeader
                title="UK"
                subTitle="Daily"
                headerStyles={{
                    backgroundColor: '#7D0068',
                    textColorPrimary: '#007ABC',
                    textColorSecondary: '#F3C100',
                }}
            />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
})
