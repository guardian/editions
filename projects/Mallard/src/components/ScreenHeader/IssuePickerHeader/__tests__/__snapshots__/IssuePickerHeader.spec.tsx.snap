// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssuePickerHeader should match the altered style by the prop headerStyles 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#052962",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "paddingVertical": 10,
      },
      Object {
        "backgroundColor": "#7D0068",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-end",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        },
        Array [
          Object {
            "marginTop": 20,
          },
          Object {
            "height": 49,
          },
        ],
        undefined,
      ]
    }
  >
    <View>
      <View
        style={
          Object {
            "paddingLeft": 14,
            "width": 90,
            "zIndex": 2,
          }
        }
      >
        <View
          accessibilityHint="Navigates to the settings screen"
          accessibilityLabel="Settings button"
          accessibilityRole="button"
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 10,
              "left": 10,
              "right": 10,
              "top": 10,
            }
          }
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 999,
                  "height": 42,
                  "justifyContent": "center",
                  "paddingHorizontal": 21,
                },
                Object {
                  "backgroundColor": undefined,
                  "borderColor": "#FFFFFF",
                  "borderWidth": 1,
                },
                Object {
                  "alignItems": "center",
                  "aspectRatio": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 0,
                  "width": 42,
                },
                undefined,
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "fontFamily": "GuardianIcons-Regular",
                    "fontSize": 20,
                    "lineHeight": 20,
                  },
                  Array [
                    Object {
                      "color": "#FFFFFF",
                    },
                    undefined,
                  ],
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
          Object {
            "width": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "row",
            },
            Object {
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "marginRight": 14,
            }
          }
        >
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View>
                <Text
                  style={
                    Array [
                      Object {
                        "flexShrink": 0,
                        "fontFamily": "GTGuardianTitlepiece-Bold",
                        "fontSize": 24,
                        "lineHeight": 26,
                      },
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "marginTop": -2,
                        },
                        undefined,
                        Object {
                          "flexShrink": 0,
                          "fontFamily": "GHGuardianHeadline-Regular",
                          "fontSize": 24,
                          "lineHeight": 26,
                        },
                        Object {
                          "color": "#007ABC",
                        },
                      ],
                    ]
                  }
                >
                  UK
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "flexShrink": 0,
                        "fontFamily": "GTGuardianTitlepiece-Bold",
                        "fontSize": 24,
                        "lineHeight": 26,
                      },
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "marginTop": -2,
                        },
                        Object {
                          "color": "#ffe500",
                        },
                        undefined,
                        Object {
                          "color": "#F3C100",
                        },
                      ],
                    ]
                  }
                >
                  Daily
                </Text>
              </View>
            </View>
          </RNGestureHandlerButton>
        </View>
        <View
          style={
            Object {
              "paddingRight": 14,
            }
          }
        >
          <View
            accessibilityHint="Returns to the edition"
            accessibilityLabel="Close button"
            accessibilityRole="button"
            accessible={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 10,
                "left": 10,
                "right": 10,
                "top": 10,
              }
            }
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 999,
                    "height": 42,
                    "justifyContent": "center",
                    "paddingHorizontal": 21,
                  },
                  Object {
                    "backgroundColor": "#ffe500",
                  },
                  Object {
                    "alignItems": "center",
                    "aspectRatio": 1,
                    "justifyContent": "center",
                    "paddingHorizontal": 0,
                    "width": 42,
                  },
                  undefined,
                ]
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "fontFamily": "GuardianIcons-Regular",
                      "fontSize": 20,
                      "lineHeight": 20,
                    },
                    Array [
                      Object {
                        "color": "#121212",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`IssuePickerHeader should match the default style 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#052962",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "paddingVertical": 10,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-end",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        },
        Array [
          Object {
            "marginTop": 20,
          },
          Object {
            "height": 49,
          },
        ],
        undefined,
      ]
    }
  >
    <View>
      <View
        style={
          Object {
            "paddingLeft": 14,
            "width": 90,
            "zIndex": 2,
          }
        }
      >
        <View
          accessibilityHint="Navigates to the settings screen"
          accessibilityLabel="Settings button"
          accessibilityRole="button"
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 10,
              "left": 10,
              "right": 10,
              "top": 10,
            }
          }
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 999,
                  "height": 42,
                  "justifyContent": "center",
                  "paddingHorizontal": 21,
                },
                Object {
                  "backgroundColor": undefined,
                  "borderColor": "#FFFFFF",
                  "borderWidth": 1,
                },
                Object {
                  "alignItems": "center",
                  "aspectRatio": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 0,
                  "width": 42,
                },
                undefined,
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "fontFamily": "GuardianIcons-Regular",
                    "fontSize": 20,
                    "lineHeight": 20,
                  },
                  Array [
                    Object {
                      "color": "#FFFFFF",
                    },
                    undefined,
                  ],
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
          Object {
            "width": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "row",
            },
            Object {
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "marginRight": 14,
            }
          }
        >
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View>
                <Text
                  style={
                    Array [
                      Object {
                        "flexShrink": 0,
                        "fontFamily": "GTGuardianTitlepiece-Bold",
                        "fontSize": 24,
                        "lineHeight": 26,
                      },
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "marginTop": -2,
                        },
                        undefined,
                        Object {
                          "flexShrink": 0,
                          "fontFamily": "GHGuardianHeadline-Regular",
                          "fontSize": 24,
                          "lineHeight": 26,
                        },
                        Object {},
                      ],
                    ]
                  }
                >
                  UK
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "flexShrink": 0,
                        "fontFamily": "GTGuardianTitlepiece-Bold",
                        "fontSize": 24,
                        "lineHeight": 26,
                      },
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "marginTop": -2,
                        },
                        Object {
                          "color": "#ffe500",
                        },
                        undefined,
                        Object {},
                      ],
                    ]
                  }
                >
                  Daily
                </Text>
              </View>
            </View>
          </RNGestureHandlerButton>
        </View>
        <View
          style={
            Object {
              "paddingRight": 14,
            }
          }
        >
          <View
            accessibilityHint="Returns to the edition"
            accessibilityLabel="Close button"
            accessibilityRole="button"
            accessible={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 10,
                "left": 10,
                "right": 10,
                "top": 10,
              }
            }
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 999,
                    "height": 42,
                    "justifyContent": "center",
                    "paddingHorizontal": 21,
                  },
                  Object {
                    "backgroundColor": "#ffe500",
                  },
                  Object {
                    "alignItems": "center",
                    "aspectRatio": 1,
                    "justifyContent": "center",
                    "paddingHorizontal": 0,
                    "width": 42,
                  },
                  undefined,
                ]
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "fontFamily": "GuardianIcons-Regular",
                      "fontSize": 20,
                      "lineHeight": 20,
                    },
                    Array [
                      Object {
                        "color": "#121212",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`IssuePickerHeader should match the default style with a subTitle 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#052962",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "paddingVertical": 10,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-end",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        },
        Array [
          Object {
            "marginTop": 20,
          },
          Object {
            "height": 49,
          },
        ],
        undefined,
      ]
    }
  >
    <View>
      <View
        style={
          Object {
            "paddingLeft": 14,
            "width": 90,
            "zIndex": 2,
          }
        }
      >
        <View
          accessibilityHint="Navigates to the settings screen"
          accessibilityLabel="Settings button"
          accessibilityRole="button"
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 10,
              "left": 10,
              "right": 10,
              "top": 10,
            }
          }
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 999,
                  "height": 42,
                  "justifyContent": "center",
                  "paddingHorizontal": 21,
                },
                Object {
                  "backgroundColor": undefined,
                  "borderColor": "#FFFFFF",
                  "borderWidth": 1,
                },
                Object {
                  "alignItems": "center",
                  "aspectRatio": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 0,
                  "width": 42,
                },
                undefined,
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "fontFamily": "GuardianIcons-Regular",
                    "fontSize": 20,
                    "lineHeight": 20,
                  },
                  Array [
                    Object {
                      "color": "#FFFFFF",
                    },
                    undefined,
                  ],
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
          Object {
            "width": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "row",
            },
            Object {
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "marginRight": 14,
            }
          }
        >
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View>
                <Text
                  style={
                    Array [
                      Object {
                        "flexShrink": 0,
                        "fontFamily": "GTGuardianTitlepiece-Bold",
                        "fontSize": 24,
                        "lineHeight": 26,
                      },
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "marginTop": -2,
                        },
                        undefined,
                        Object {
                          "flexShrink": 0,
                          "fontFamily": "GHGuardianHeadline-Regular",
                          "fontSize": 24,
                          "lineHeight": 26,
                        },
                        Object {},
                      ],
                    ]
                  }
                >
                  Recent
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "flexShrink": 0,
                        "fontFamily": "GTGuardianTitlepiece-Bold",
                        "fontSize": 24,
                        "lineHeight": 26,
                      },
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "marginTop": -2,
                        },
                        Object {
                          "color": "#ffe500",
                        },
                        undefined,
                        Object {},
                      ],
                    ]
                  }
                >
                  Editions
                </Text>
              </View>
            </View>
          </RNGestureHandlerButton>
        </View>
        <View
          style={
            Object {
              "paddingRight": 14,
            }
          }
        >
          <View
            accessibilityHint="Returns to the edition"
            accessibilityLabel="Close button"
            accessibilityRole="button"
            accessible={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 10,
                "left": 10,
                "right": 10,
                "top": 10,
              }
            }
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 999,
                    "height": 42,
                    "justifyContent": "center",
                    "paddingHorizontal": 21,
                  },
                  Object {
                    "backgroundColor": "#ffe500",
                  },
                  Object {
                    "alignItems": "center",
                    "aspectRatio": 1,
                    "justifyContent": "center",
                    "paddingHorizontal": 0,
                    "width": 42,
                  },
                  undefined,
                ]
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "fontFamily": "GuardianIcons-Regular",
                      "fontSize": 20,
                      "lineHeight": 20,
                    },
                    Array [
                      Object {
                        "color": "#121212",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
